wa/framework/execution.py

```diff
@@ -657,6 +657,9 @@ def check_job(self, job):
                 self.logger.error(msg.format(job.id, job.iteration, job.status))
                 self.context.failed_jobs += 1
                 self.send(signal.JOB_FAILED)
+                if rc.bail_on_job_failure:
+                    raise WorkloadError('Job {} failed, bailing.'.format(job.id))
```

Comment:
This shouldn't be a `WorkloadError`. `ExecutionError` is probably the most appropriate.